src/basic/REDUCE_STRUCT.hpp

```diff
@@ -0,0 +1,135 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-21, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// REDUCE_STRUCT kernel reference implementation:
+///
+/// Real_type xsum; Real_type ysum;
+/// Real_type xmin; Real_type ymin;
+/// Real_type xmax; Real_type ymax;
```

Comment:
How should these be initialized? In the implementations the min is initialized to 0, but then it can never find a min greater than 0.

Reply:
> How should these be initialized? In the implementations the min is initialized to 0, but then it can never find a min greater than 0.

@MrBurmark that's a good observation. The array initialization in REDUCE_STRUCT::setUp( ) will always set the first elements to zero. We should probably change that or initialize the reduction variables to something else to catch issues.

Reply:
I think that is a good idea